### ISSUE TITLE
Error when checking target: expected dense_2 to have shape (None, 1) but got array with shape (3230, 7)

### ISSUE THREAD

**Issue by ghost** [keras-team/keras#8755]
rain = pd.read_csv("test.csv")
test = pd.read_csv("testprivate.csv")

print("train:", train)
print("test:", test)

#I Choose from train dataset the emotion
Y_train = train["emotion"]
print("Y_train:", Y_train)

# Drop 'label' column
X_train = train.drop(labels = ["emotion"],axis = 1)
print("X_pixels:", X_train)

Y_test = test["emotion"]
print("Y_train:", Y_train)

# Drop 'label' column
X_test = test.drop(labels = ["emotion"],axis = 1)
print("X_pixels:", X_train)

g = sns.countplot(Y_train)
gtest = sns.countplot(Y_test)

Y_train.value_counts()
Y_test.value_counts()

plt.show(g)
plt.show(gtest)

# Check the data
print(X_train.isnull().any().describe())
print(test.isnull().any().describe())

# Normalize the data
X_train = X_train.astype('float32')/ 255.0

test = test.astype('float32')/ 255.0

# Reshape image in 3 dimensions (height = 48px, width = 48px , canal = 1)
X_train = X_train.values.reshape(-1, 48, 48, 1)
#test = test.values.reshape(-1,48, 48, 1)

# Encode labels to one hot vectors (ex : 2 -> [0,0,1,0,0,0,0,0,0,0])
Y_train = to_categorical(Y_train, num_classes = 7)

# Set the random seed
random_seed = 2

# Split the train and the validation set for the fitting
X_train, X_val, Y_train, Y_val = train_test_split(X_train, Y_train, test_size = 0.1, random_state=random_seed)

model = Sequential()

model.add(Conv2D(filters = 32, kernel_size = (5,5), padding = 'Same',
                 activation ='relu', input_shape = (48, 48, 1)))
model.add(Conv2D(filters = 32, kernel_size = (5,5),padding = 'Same',
                 activation ='relu'))
model.add(MaxPool2D(pool_size=(2,2)))
model.add(Dropout(0.25))

model.add(Conv2D(filters = 64, kernel_size = (3,3),padding = 'Same',
                 activation ='relu'))
model.add(Conv2D(filters = 64, kernel_size = (3,3),padding = 'Same',
                 activation ='relu'))
model.add(MaxPool2D(pool_size=(2,2), strides=(2,2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(256, activation = "relu"))
model.add(Dropout(0.5))
model.add(Dense(10, activation = "sigmoid"))
print(model.summary())

# Define the optimizer
optimizer = RMSprop(lr=0.001, rho=0.9, epsilon=1e-08, decay=0.0)

# Compile the model
model.compile(optimizer = optimizer , loss = "categorical_crossentropy", metrics=["accuracy"])

# Set a learning rate annealer
learning_rate_reduction = ReduceLROnPlateau(monitor='val_acc',
                                            patience=3,
                                            verbose=1,
                                            factor=0.5,
                                            min_lr=0.00001)

epochs = 1 
batch_size = 86

model.fit(X_train, Y_train, batch_size = batch_size, epochs = epochs, validation_data = (X_val, Y_val), verbose = 2)

**Comment by Dref360** [comment]
Hi!
This question seems best suited for StackOverflow. Could you post your issue there? Also, remember to close your issue if you do so!
Thanks!
